### PR TITLE
implemented correct k8s validation

### PR DESCRIPTION
**Pull request by zvonkok**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -28,19 +28,21 @@ path = "src/bin/validate.rs" # Path to the source file
...
+clap = "4.5.4"
 serde_json = "1.0"
 jsonschema = "0.18.0"
 once_cell = "1.8.0"
 notify = "6.1.1"
 serde = "1.0.131"
 serde_derive = "1.0.131"
 libc = "0.2.112"
-nix = "0.29.0"
+nix = "0.24.2"
```
Comment (zvonkok) [inline Cargo.toml:39]: Downgrade for kata-agent build

File: Cargo.toml [hunk 2]
```diff
@@ -28,19 +28,21 @@ path = "src/bin/validate.rs" # Path to the source file
...
 lazy_static = "1.2"
 path-clean = "1.0.1"
 serde_yaml = "0.9.33"
-semver = "1.0.23"
```
Comment (zvonkok) [inline Cargo.toml:43]: Downgrade for kata-agent build

File: src/annotations.rs [hunk 3]
```diff
@@ -169,8 +169,8 @@ mod tests {
                 .to_string(),
         );
 
-        assert!(parse_annotations(cdi_devices.clone()).is_ok());
-        let (keys, devices) = parse_annotations(cdi_devices).unwrap();
+        assert!(parse_annotations(&cdi_devices).is_ok());
+        let (keys, devices) = parse_annotations(&cdi_devices).unwrap();
```
Comment (zvonkok) [inline src/annotations.rs:173]: Reference for using in kata-agent we do not need to own the annotations.

File: src/internal/validation/k8s/objectmeta.rs [hunk 4]
```diff
@@ -1,7 +1,105 @@
...
+
+        assert!(validate_annotations(&annotations, path).is_ok());
+
+        let mut large_annotations = BTreeMap::new();
+        let long_value = "CDI".repeat(super::TOTAL_ANNOTATION_SIZE_LIMIT + 1);
+        large_annotations.insert("CDIKEY".to_string(), long_value);
+        assert!(validate_annotations(&large_annotations, path).is_err());
+
+        let mut invalid_annotations = BTreeMap::new();
+        invalid_annotations.insert("inv$$alid_CDIKEY".to_string(), "inv$$alid_CDIVAL".to_string());
```
Comment (zvonkok) [inline src/internal/validation/k8s/objectmeta.rs:73]: `invalid_CDIKEY` is actually a valid name, added `$$` which is not valid

File: src/internal/validation/k8s/validation.rs [hunk 5]
```diff
@@ -1 +1,211 @@
...
+        }
+        _ => {
+            return vec![format!(
+			"a qualified name {} with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')",
+			regex_error(QUALIFIED_NAME_ERR_MSG, QUALIFIED_NAME_FMT, &["MyName", "my.name", "123-abc"])
+		    )];
+        }
+    }
+
+    println!("### TEST");
```
Comment (Apokleos) [inline src/internal/validation/k8s/validation.rs]: debug info needs be removed

File: src/internal/validation/k8s/validation.rs [hunk 6]
```diff
@@ -1 +1,211 @@
...
+
+    if name.is_empty() {
+        errs.push(format!("name part {}", empty_error()));
+    } else if name.len() > QUALIFIED_NAME_MAX_LENGTH {
+        errs.push(format!(
+            "name part {}",
+            max_len_error(QUALIFIED_NAME_MAX_LENGTH)
+        ));
+    }
+    println!("### TEST name {}", name);
```
Comment (Apokleos) [inline src/internal/validation/k8s/validation.rs]: remove debug info too

**Review by Apokleos — COMMENTED** [review]
Hi Zvonko, this PR almost is fine, but two debug info need to be removed and we need do cargo fmt&clippy

**Pull request merged**